src/net.h

```diff
@@ -136,6 +136,33 @@ class CConnman
 
     bool ForNode(NodeId id, std::function<bool(CNode* pnode)> func);
 
+    template <typename... Args>
```

Comment:
PushMessageWithVersionAndFlag could be private I suppose. It's only used by the others.

PushMessageWithFlag is the same as before.

PushMessageWithVersion is used explicitly before the handshake is complete, before we know what upgraded version to use.